src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -75,6 +75,24 @@ internal ObjectReader(Stream stream, ISurrogateSelector? selector, StreamingCont
             _formatterEnums = formatterEnums;
         }
 
+        [RequiresDynamicCode(ObjectReaderUnreferencedCodeMessage)]
+        [RequiresUnreferencedCode("Types might be removed")]
+        private static IDisposable? StartDeserialization()
+        {
+            MethodInfo? targetMethod = typeof(SerializationInfo).GetMethod(
+                "StartDeserialization",
+                BindingFlags.Public | BindingFlags.Static,
+                Type.EmptyTypes);
+
+            // It might have been removed in AoT.
```

Comment:
This does not look right. Trimming and native AOT recognizes the above pattern as intrinsic. What is the problem that this is trying to solve?

Reply:
The call has to switch to reflection because StartDeserialization is not exposed via any of the reference assemblies.

If trimming and AOT see this particular form of reflection as the same as real method invocation, I can certainly remove the comment and change the throw to an assert.

Reply:
Yes, the comment and the Requires... attributes on the method should be deleted. 

Up to you whether you want to keep the unreachable explicit throw as defensive programming or whether assert is enough.